File: app/src/main/java/com/example/flashcardlab2/MainActivity.java
package com.example.flashcardlab2;

import android.app.Activity;
import android.content.Intent;
import android.content.res.ColorStateList;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.animation.CycleInterpolator;
import android.view.animation.TranslateAnimation;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends Activity {

    private int currentQuestionNumber = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button saveButton = findViewById(R.id.saveButton);
        EditText questionInput = findViewById(R.id.questionInput);
        EditText answer1Input = findViewById(R.id.answer1Input);

        EditText answer2Input = findViewById(R.id.answer2Input);
        EditText answer3Input = findViewById(R.id.answer3Input);
        answer2Input.setVisibility(View.GONE);
        answer3Input.setVisibility(View.GONE);


        questionInput.setText("Quelle est la capitale de la France ?");
        questionInput.setEnabled(false);
        answer1Input.setText("");

        saveButton.setOnClickListener(v -> {
            String userAnswer = answer1Input.getText().toString().trim();


            if (currentQuestionNumber == 1) {
                if (userAnswer.equalsIgnoreCase("Paris")) {

                    answer1Input.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#4CAF50")));
                    Toast.makeText(MainActivity.this, "Good! Prêt pour la suite ?", Toast.LENGTH_SHORT).show();

                    currentQuestionNumber = 2;


                    new Handler().postDelayed(() -> {
                        answer1Input.setBackgroundTintList(ColorStateList.valueOf(Color.GRAY));
                        questionInput.setText("Quel est le plus grand pays du monde ?");
                        answer1Input.setText("");
                    }, 1000);

                } else {

                    handleWrongAnswer(answer1Input);
                }
            }

            else if (currentQuestionNumber == 2) {
                if (userAnswer.equalsIgnoreCase("Russie")) {
                    // --- BON ---
                    answer1Input.setBackgroundTintList(ColorStateList.valueOf(Color.parseColor("#4CAF50")));
                    Toast.makeText(MainActivity.this, "Good! Tout est correct !", Toast.LENGTH_SHORT).show();

                    Intent data = new Intent();
                    data.putExtra("ANSWER_KEY", userAnswer);
                    setResult(RESULT_OK, data);
                    finish();

                } else {

                    handleWrongAnswer(answer1Input);
                }
            }
        });
    }

    private void handleWrongAnswer(EditText answerInput) {
        answerInput.setBackgroundTintList(ColorStateList.valueOf(Color.RED));

        Toast.makeText(MainActivity.this, "Echec", Toast.LENGTH_SHORT).show();

        TranslateAnimation shake = new TranslateAnimation(0, 10, 0, 0);
        shake.setDuration(500);
        shake.setInterpolator(new CycleInterpolator(7));
        answerInput.startAnimation(shake);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                answerInput.setText("");
                answerInput.setBackgroundTintList(ColorStateList.valueOf(Color.GRAY));
            }
        }, 1000);
    }
}

File: app/src/main/java/com/example/flashcardlab2/MainActivity.kt
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.flashcardlab2.R
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.questionInput )) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        val questionInput = findViewById<EditText>(R.id.questionInput)
        val answer1Input = findViewById<EditText>(R.id.answer1Input)
        val answer2Input = findViewById<EditText>(R.id.answer2Input)
        val answer3Input = findViewById<EditText>(R.id.answer3Input)
        val saveButton = findViewById<Button>(R.id.saveButton)


        saveButton.setOnClickListener {

            val question = questionInput.text.toString()
            val answer1 = answer1Input.text.toString()
            val answer2 = answer2Input.text.toString()
            val answer3 = answer3Input.text.toString()


            if (question.isBlank() || answer3.isBlank()) {
                Toast.makeText(
                    this,
                    "Please enter both a question and the correct answer!",
                    Toast.LENGTH_SHORT
                ).show()
                return@setOnClickListener
            }


            Snackbar.make(saveButton, "Card Saved! Question: $question", Snackbar.LENGTH_LONG)
                .show()

        }
    }

    private fun findViewById(id: Any): View {
        return TODO("Provide the return value")
    }
}